packages/sdk/src/oracle/README.md

```diff
@@ -0,0 +1,77 @@
+# Oracle Client
+
+Interact with UMA's Optimistic Oracle using an eventful client designed for a frontend dapp with global state management.
+
+# Quick Start
+
+Currently theres a single factory which creates the client and allows you to subscribe to state changes.
+
+```js
+import {oracle} from '@uma/sdk'
+
+// create your configuration object with known addresses
+type Config = oracle.types.state.Config
+
+// creating a config for mainnet usage add any additional chains you want
+export const config = {
+  chains: {
+    1:{
+      chainId: 1,
+      multicall2Address: process.env.multicall2Address,
+      optimisticOracleAddress: process.env.optimisticOracleAddress,
+      providerUrl: process.env.CUSTOM_NODE_URL,
+    },
+  },
+};
+
+let state:oracle.types.state.State = {}
+function changeHandler(nextState:oracle.types.state.State, prevState:oracle.types.state.State){
+ // dispatch to state manager. Recommended to have its own slice of state.
+ state = nextState
+}
+
+const client = Factory(config, changeHandler);
+// client contains the following classes and calls:
+//{
+//  store: { read, write, get} // get and set all parts of state
+//  update: {// all update calls} // update parts of state on chain
+//  setUser: // set the currently logged in user
+//  setActiveRequest: // set the request page the user is on
+//}
+
+const requester = "0xb8b3583f143b3a4c2aa052828d8809b0818a16e9",
+const identifier = "0x554D415553440000000000000000000000000000000000000000000000000000"
+const timestamp = 1638453600
+const ancillaryData = "0x747761704C656E6774683A33363030",
+
+const account = "0x9A8f92a830A5cB89a3816e3D267CB7791c16b04D";
+const chainId = ChainId.MAINNET;
+
+// currently the order of calls is important
+client.setActiveRequest(requester,identifier,timestamp,ancillaryData)
```

Comment:
Interesting -- why are we having the client have a stateful "active request" OOC? I'm trying to think about this purely from a user's perspective (without considering what's happening under the hood).

Two alternatives that I could see being a little simpler to understand/use:
1. Having an async method on the client to get the raw request state `client.getRequest(...)`, since that's all you can really do with a particular request. This would drop the statefulness in the client, which seems like it might be a bit annoying to deal with if you reuse the same client across a dapp.
2. Similar to 1, but maybe a little closer to what's already here: if you want request to be a stateful object, you could have a `client.getRequest(...)` function that returns a request object that you can then use to interact with the request in a more dynamic way. Something like:
```ts
const request = client.getRequest(requester,identifier,timestamp,ancillaryData);

// Not exactly sure how the user should trigger updates, as that's the part that's very application/system specific.
await request.update();

// Some way to directly get the request state would be ideal.
const state = request.read().state();

await request.write().dispute(...);

...
```